### PR TITLE
Connect to VNET

### PR DESCRIPTION
**Pull request by djorgensendk**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
     uses: Energinet-DataHub/.github/.github/workflows/terraform-validate.yml@3.0.2
     with:
       TERRAFORM_WORKING_DIR_PATH: './build/infrastructure/main'
-      TERRAFORM_VERSION: '1.0.10'
+      TERRAFORM_VERSION: '1.1.2'
```
Comment (Renetnielsen) [inline .github/workflows/ci.yml:47]: Since were at it, lets use 1.1.5 (newest)

File: build/infrastructure/main/providers.tf [hunk 2]
```diff
@@ -12,11 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 terraform {
-  required_version = "=1.0.10"
+  required_version = "=1.1.2"
```
Comment (Renetnielsen) [inline build/infrastructure/main/providers.tf:15]: Since were at it, lets use 1.1.5 (newest)

File: build/infrastructure/main/subnets.tf [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+module "snet_internal_private_endpoints" {
+  source                                          = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/subnet?ref=6.0.0"
+  name                                            = "internal-private-endpoints"
+  project_name                                    = var.domain_name_short
+  environment_short                               = var.environment_short
+  environment_instance                            = var.environment_instance
+  resource_group_name                             = data.azurerm_key_vault_secret.vnet_internal_resource_group_name.value
+  virtual_network_name                            = data.azurerm_key_vault_secret.vnet_internal_name.value
+  address_prefixes                                = [var.subnet_internal_private_endpoints_address_space]
```
Comment (Renetnielsen) [inline build/infrastructure/main/subnets.tf:23]: Please indent

```
address_prefixes = [
  var.subnet_internal_private_endpoints_address_space
]
```

Just to follow what we did elsewhere.

As far as i remember Terraform will minify it on build, so it does not matter how its written in plain files.

File: build/infrastructure/main/subnets.tf [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+module "snet_external_private_endpoints" {
+  source                                          = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/subnet?ref=6.0.0"
+  name                                            = "external-private-endpoints"
+  project_name                                    = var.domain_name_short
+  environment_short                               = var.environment_short
+  environment_instance                            = var.environment_instance
+  resource_group_name                             = data.azurerm_key_vault_secret.vnet_internal_resource_group_name.value
+  virtual_network_name                            = data.azurerm_key_vault_secret.vnet_internal_name.value
+  address_prefixes                                = [var.subnet_external_private_endpoints_address_space]
```
Comment (Renetnielsen) [inline build/infrastructure/main/subnets.tf:36]: Please indent

```
address_prefixes = [
  var.subnet_external_private_endpoints_address_space
]
```

Just to follow what we did elsewhere.

As far as i remember Terraform will minify it on build, so it does not matter how its written in plain files.

File: build/infrastructure/main/subnets.tf [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+module "vnet_integrations_webapi" {
+  source                                          = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/subnet?ref=6.0.0"
+  name                                            = "vnet-integrations-webapi"
+  project_name                                    = var.domain_name_short
+  environment_short                               = var.environment_short
+  environment_instance                            = var.environment_instance
+  resource_group_name                             = data.azurerm_key_vault_secret.vnet_internal_resource_group_name.value
+  virtual_network_name                            = data.azurerm_key_vault_secret.vnet_internal_name.value
+  address_prefixes                                = [var.subnet_vnet_integrations_webapi_address_space]
```
Comment (Renetnielsen) [inline build/infrastructure/main/subnets.tf:48]: Please indent

```
address_prefixes = [
  var.subnet_vnet_integrations_webapi_address_space
]
```

Just to follow what we did elsewhere.

As far as i remember Terraform will minify it on build, so it does not matter how its written in plain files.

File: build/infrastructure/main/subnets.tf [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  enforce_private_link_service_network_policies   = true
+
+  # Delegate the subnet to "Microsoft.Web/serverFarms"
+  delegations =  [{
+   name = "delegation"
+   service_delegation_name    = "Microsoft.Web/serverFarms"
+   service_delegation_actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
```
Comment (Renetnielsen) [inline build/infrastructure/main/subnets.tf:55]: Please indent

```
service_delegation_actions = [
  "Microsoft.Network/virtualNetworks/subnets/action"
]
```

Just to follow what we did elsewhere.

As far as i remember Terraform will minify it on build, so it does not matter how its written in plain files.

File: build/infrastructure/main/subnets.tf [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+module "vnet_integrations_functions" {
+  source                                          = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/subnet?ref=6.0.0"
+  name                                            = "vnet-integrations-functions"
+  project_name                                    = var.domain_name_short
+  environment_short                               = var.environment_short
+  environment_instance                            = var.environment_instance
+  resource_group_name                             = data.azurerm_key_vault_secret.vnet_internal_resource_group_name.value
+  virtual_network_name                            = data.azurerm_key_vault_secret.vnet_internal_name.value
+  address_prefixes                                = [var.subnet_vnet_integrations_functions_address_space]
```
Comment (Renetnielsen) [inline build/infrastructure/main/subnets.tf:67]: Please indent

```
address_prefixes = [
  var.subnet_vnet_integrations_functions_address_space
]
```

Just to follow what we did elsewhere.

As far as i remember Terraform will minify it on build, so it does not matter how its written in plain files.

File: build/infrastructure/main/subnets.tf [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  enforce_private_link_service_network_policies   = true
+
+  # Delegate the subnet to "Microsoft.Web/serverFarms"
+  delegations =  [{
+   name = "delegation"
+   service_delegation_name    = "Microsoft.Web/serverFarms"
+   service_delegation_actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
```
Comment (Renetnielsen) [inline build/infrastructure/main/subnets.tf:74]: Please indent

```
service_delegation_actions = [
 "Microsoft.Network/virtualNetworks/subnets/action"
]
```

Just to follow what we did elsewhere.

As far as i remember Terraform will minify it on build, so it does not matter how its written in plain files.

**Review by Renetnielsen — APPROVED** [review]
LGTM, minor indention changes

File: build/infrastructure/main/subnets.tf (hunk 8 above)
Comment (djorgensendk) [reply build/infrastructure/main/subnets.tf]: done

File: build/infrastructure/main/subnets.tf (hunk 7 above)
Comment (djorgensendk) [reply build/infrastructure/main/subnets.tf:75]: done

File: build/infrastructure/main/subnets.tf (hunk 6 above)
Comment (djorgensendk) [reply build/infrastructure/main/subnets.tf]: done

File: build/infrastructure/main/subnets.tf (hunk 5 above)
Comment (djorgensendk) [reply build/infrastructure/main/subnets.tf]: done

File: build/infrastructure/main/subnets.tf (hunk 4 above)
Comment (djorgensendk) [reply build/infrastructure/main/subnets.tf]: done

File: build/infrastructure/main/subnets.tf (hunk 3 above)
Comment (djorgensendk) [reply build/infrastructure/main/subnets.tf]: done

File: build/infrastructure/main/providers.tf (hunk 2 above)
Comment (djorgensendk) [reply build/infrastructure/main/providers.tf]: done

File: .github/workflows/ci.yml (hunk 1 above)
Comment (djorgensendk) [reply .github/workflows/ci.yml]: done

**Review by Renetnielsen — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]

**Review by Renetnielsen — DISMISSED** [review]

**Pull request merged**